xos.h

```diff
@@ -101,36 +102,44 @@ namespace xos
     int lookahead_;
   };
 
-} // namespace xos
+  class StrAST
+  {
+  public:
+    std::string StrVal;
```

Comment:
Members should be `private` and end with an underscore. See "Class data members" in https://google.github.io/styleguide/cppguide.html